vita3k/gui/src/controllers_dialog.cpp

```diff
@@ -186,22 +186,30 @@ static void swap_controller_ports(CtrlState &state, int source_port, int dest_po
     auto dest_controller_it = std::find_if(state.controllers.begin(), state.controllers.end(),
         [&](const auto &pair) { return pair.second.port == (dest_port + 1); });
 
-    // Check that both controllers exist
-    if ((source_controller_it == state.controllers.end()) || (dest_controller_it == state.controllers.end())) {
-        LOG_ERROR("Unable to find one or more controllers on the specified ports.");
+    // Check that source controllers exist
+    if (source_controller_it == state.controllers.end()) {
+        LOG_ERROR("Unable to find controller on source port {}", source_port);
         return;
     }
 
-    LOG_INFO("Controller on source port {} {} swapped with controller on destination port {} {}", source_port, state.controllers_name[source_port], dest_port, state.controllers_name[dest_port]);
+    if (dest_controller_it == state.controllers.end()) {
+        LOG_INFO("Controller on source port {} {} assigned to destination port {}", source_port, source_controller_it->second.name, dest_port);
 
-    // Swap controllers port and player index
-    SDL_GameControllerSetPlayerIndex(source_controller_it->second.controller.get(), dest_port);
-    SDL_GameControllerSetPlayerIndex(dest_controller_it->second.controller.get(), source_port);
-    std::swap(source_controller_it->second.port, dest_controller_it->second.port);
+        // Assign controller to destination port
+        SDL_GameControllerSetPlayerIndex(source_controller_it->second.controller.get(), dest_port);
+        source_controller_it->second.port = dest_port + 1;
+        std::swap(state.free_ports[source_port], state.free_ports[dest_port]);
+    } else {
+        LOG_INFO("Controller on source port {} {} swapped with controller on destination port {} {}", source_port, source_controller_it->second.name, dest_port, dest_controller_it->second.name);
 
-    // Swap controller names and motion support
-    std::swap(state.controllers_name[source_port], state.controllers_name[dest_port]);
-    std::swap(state.controllers_has_motion_support[source_port], state.controllers_has_motion_support[dest_port]);
+        // Swap controllers port and player index
+        SDL_GameControllerSetPlayerIndex(source_controller_it->second.controller.get(), dest_port);
+        SDL_GameControllerSetPlayerIndex(dest_controller_it->second.controller.get(), source_port);
+        std::swap(source_controller_it->second.port, dest_controller_it->second.port);
+
+        // Swap controller names and motion support
+        std::swap(state.free_ports[source_port], state.free_ports[dest_port]);
```

Comment:
This just swap two booleans, it has nothing to do with swapping controller names and motion support

Reply:
I didn't update the existing comments while modifying the logic.

And looking at it again, since the free_ports values on both sides are identical when entering that part of the conditional statement, that code is unnecessary.
I think those lines remained because they were written before the conditional branching. I'll remove that part. Thank you.